src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -83,19 +82,22 @@ public List<EnrollmentDTO> getTranscript(@RequestParam("studentId") int studentI
     // user must be student
     // studentId will be temporary until Login security is implemented
    @GetMapping("/enrollments")
+   @PreAuthorize("hasAuthority('SCOPE_ROLE_STUDENT')")
    public List<EnrollmentDTO> getSchedule(
-           @RequestParam("year") int year,
-           @RequestParam("semester") String semester,
-           @RequestParam("studentId") int studentId) {
+       @RequestParam("year") int year,
+       @RequestParam("semester") String semester,
+       Principal principal) {
+
+       User student = userRepository.findByEmail(principal.getName());
 
        List<EnrollmentDTO> schedule = new ArrayList<>();
-       List<Enrollment> enrollments = enrollmentRepository.findByYearAndSemesterOrderByCourseId(year, semester, studentId);
+       List<Enrollment> enrollments = enrollmentRepository.findByYearAndSemesterOrderByCourseId(year, semester, student.getId());
        //we are not checking for empty enrollments/ enrollments not found
        for (Enrollment e : enrollments) {
 
            if (!e.getStudent().getType().equals("STUDENT")) {
                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "user is not a student");
-           } else if (studentId != e.getStudent().getId()) {
+           } else if (student.getId() != e.getStudent().getId()) {
                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "student id does not match");
```

Comment:
I am not sure if this is necessary if we retrieved enrollments with the student's Id, and we used  ```SCOPE_ROLE_STUDENT```